### ISSUE TITLE
Crashes badly with LSP-plugins

### ISSUE THREAD

**Issue by oldcastlehq** [rncbc/qtractor#396]
Hi,

Qtractor crashes when I try to add the LSP-plugins Multi-Sampler x12 DirectOut plugin (CLAP).

Any idea why?

**Comment by rncbc** [comment]
no idea; care to have a debug build and get a backtrace?

**Comment by rncbc** [comment]
fixed in qtractor >= 0.9.33.35git.[d01c8d](https://github.com/rncbc/qtractor/commit/d01c8dc)

thanks

**Comment by oldcastlehq** [comment]
It worked for me.

**Comment by oldcastlehq** [comment]
It may be related, since now all Mixer x4/8/16 Stereo/Mono plugins crash Qtractor.

**Comment by rncbc** [comment]
you have no use for plugins that present more inputs than audio outputs in qtractor;

in a word: don't use them: they'll never work as intended

if you insist, you may use something like Carla instead.

**Comment by oldcastlehq** [comment]
Ok... I don't understand here. So, some plugins from LSP have more audio inputs than audio outputs in qtractor. Why is that? Is the LSP desgin or a Qtractor limitation?

**Comment by rncbc** [comment]
both. 

1. LSP design: plugins that have ins > outs  are simply useless in qtractor--don't ever rely on them; they won't ever work as expected anyway; although reasons why it crashes  goes beyhond my current understanding.
2. Qtractor limitation: yes, it only works correctly for a number of audio channels that match the current track or output bus count; anything else will get lost or just non-operational, if not dysfunctional.

**Comment by sadko4u** [comment]
At least there is `CLAP_AUDIO_PORT_IS_MAIN` constant that may give a hint to the Host that this audio port should be used as a primary for the track/bus.

**Comment by sadko4u** [comment]
@rncbc

> plugins that have ins > outs are simply useless in qtractor--don't ever rely on them;

How do you deal with sidechain then?

**Comment by oldcastlehq** [comment]
> @oldcastlehq Oh, I forgot to ask which format of plugins do you try to load in Qtractor?

CLAP

**Comment by rncbc** [comment]
> How do you deal with sidechain then?

qtractor does not support sidechain scenarios.

**Comment by rncbc** [comment]
> At least there is `CLAP_AUDIO_PORT_IS_MAIN` constant that may give a hint to the Host that this audio port should be used as a primary for the track/bus.

that was exactly what was there before this later fix: somehow, none of the LSP Multisampler plugins report a single CLAP_AUDIO_PORT_IS_MAIN.

**Comment by sadko4u** [comment]
> none of the LSP Multisampler plugins report a single CLAP_AUDIO_PORT_IS_MAIN

Hmm, I need to look at this.

**Comment by rncbc** [comment]
yes you should; I just reverted the said fix, just bcoz surge-xt .clap wouldn't work otherwise, sorry
